modular_skyrat/modules/primitive_cooking_additions/code/millstone.dm

```diff
@@ -66,6 +66,28 @@
 	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
 
 /obj/structure/millstone/attackby(obj/item/attacking_item, mob/user)
+	if(istype(attacking_item, /obj/item/storage/bag))
+		if(length(contents) >= maximum_contained_items)
+			balloon_alert(user, "already full")
+			return TRUE
+		
+		if(!attacking_item.contents.len)
+			balloon_alert(user, "bag empty")
+			return TRUE
+			
+		for(var/obj/item/food/grown/target_item in attacking_item.contents)
+			if(length(contents) >= maximum_contained_items)
+				break
+				
+			target_item.forceMove(src)
+		
+		if (length(contents) >= maximum_contained_items)
+			balloon_alert(user, "millstone filled")
+		else
+			balloon_alert(user, "bag emptied")
```

Comment:
Same comment as for the other one, though. What if you try to put stuff in it and nothing comes out, or there's still stuff left in your bag?

I feel like we could probably use the code for dumping things into a container using click-drag instead, which already handles almost all of that (minus I guess a check we'd have to add to it for this), no?